### ISSUE TITLE
Hex grids: implement nodes_at_top_edge, nodes_at_left_edge, etc.

### ISSUE THREAD

**Issue by gregtucker** [landlab/landlab#628]
There should also be a better way than examining every node's x and y. Methods to identify nodes on boundaries should also take into account an origin that isn't (0, 0).

**Comment by kbarnhart** [comment]
A clarification question: You are proposing six methods/attributes that would get the node IDs of the six sides of the hex grid, right? The equivalent to the nodes_at_top_edge methods for rasters often used to set boundary conditions. 

If so, I concur! I wanted this functionality a few days ago.

A semantic comment: Since the six edges do not have obvious names (unless we go with S, SE, NE, N, NW, SW) we should think carefully about how we identify them.

**Comment by gregtucker** [comment]
Yes, exactly. Hex grids can have a rectangular or hex shape, so the number of "cardinal" sides is either 2 or 4. I posted this issue as a way of procrastinating (but not losing) a good suggestion of Eric's on a current pull request...
